polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -259,3 +267,55 @@ $breakpoint: 768px;
     background: none;
   }
 }
+
+.StickyTable {
+  position: relative;
+  top: 0;
+  left: 0;
+  right: 0;
+  visibility: hidden;
+  z-index: var(--p-z-1);
+}
+
+.StickyTableHeader {
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  overflow: hidden;
+
+  &:not(.StickyTableHeader-isSticky) {
+    top: -1000px;
+    left: -1000px;
+  }
+}
+
+.StickyHeaderCell {
+  flex-shrink: 0;
+}
+
+.StickyTableColumnHeader-isScrolling {
+  box-shadow: 1px 1px 0 0 var(--p-border-divider),
+    1px 0 1px 1px rgba(63, 63, 68, 0.05), 1px 0 3px 0 rgba(63, 63, 68, 0.15);
+}
+
+.StickyTableHeadingsRow {
+  flex: 1 1 auto;
+  display: flex;
+  overflow-x: auto;
+  background-color: var(--p-surface);
+  justify-content: space-between;
+  scrollbar-width: none;
+
+  &::-webkit-scrollbar {
```

Comment:
Should we do anything else for other browsers? https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar